### PR TITLE
Balance job / service refactoring

### PR DESCRIPTION
**Pull request by polonzer**
https://lykkex.atlassian.net/browse/LWDEV-8348

### CODE REVIEW

File: src/Lykke.Job.Balances/Modules/AutofacDependencyResolver.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using Autofac;
+using Lykke.Cqrs;
+
+namespace Lykke.Job.Balances.Modules
+{
+    internal class AutofacDependencyResolver : IDependencyResolver
```
Comment (samodovdi) [inline src/Lykke.Job.Balances/Modules/AutofacDependencyResolver.cs:7]: just a note - i think we need to move this class to cqrs library, as its always copy/pasted in the projects with CQRS

File: src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+using JetBrains.Annotations;
+using Lykke.Service.Balances.Core.Domain.Wallets;
+using Lykke.Service.Balances.Core.Services.Wallets;
+using Lykke.Service.Balances.Services.Wallet.CacheModels;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace Lykke.Service.Balances.Services.Wallet
+{
+    [UsedImplicitly]
+    public class CachedCachedWalletsRepository : ICachedWalletsRepository
```
Comment (paradise) [inline src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs]: CachedCached ?

File: src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                await _cache.TryRemoveAsync(GetAllBalancesCacheKey(walletId));
+            }
+        }
+
+        public Task CacheItAsync(string walletId)
+        {
+            return GetAllAsync(walletId);
+        }
+
+        public async Task<IReadOnlyList<IWallet>> GetTotalBalancesAsync()
```
Comment (paradise) [inline src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs]: Is it needed ? Who updates "totalBalance" ?

File: src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+                GetTotalBalancesCacheKey(),
+                async () => (await _repository.GetTotalBalancesAsync())
+                    .Select(CachedWalletModel.Copy)
+                    .ToArray(),
+                slidingExpiration: _cacheExpiration,
+                log: _log);
+        }
+
+
+        private static string GetAllBalancesCacheKey(string clientId) => $":balances:{clientId}:all";
```
Comment (paradise) [inline src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs]: string walletId

File: src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        private static string GetAssetBalanceCacheKey(string clientId, string assetId) => $":balances:{clientId}:{assetId}";
```
Comment (paradise) [inline src/Lykke.Service.Balances.Services/Wallet/CachedCachedWalletsRepository.cs]: the same

**Review by paradise — APPROVED** [review]

**Review by samodovdi — APPROVED** [review]

**Pull request merged**